src/overlays/actors/ovl_Demo_Effect/z_demo_effect.c

```diff
@@ -20,121 +105,2115 @@ const ActorInit Demo_Effect_InitVars = {
+    gSPDisplayList(POLY_XLU_DISP++, fireBall);
+    CLOSE_DISPS(globalCtx->state.gfxCtx, "../z_demo_effect.c", 2723);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Demo_Effect/func_80972288.s")
+/**
+ * Draw function for the God Lgt Actors.
+ * This draws either Din, Nayru, or Farore based on the colors set in the DemoEffect struct.
+ */
+void DemoEffect_DrawGodLgt(DemoEffect* this, GlobalContext* globalCtx) {
+    s32 pad;
+    s32 pad2;
+    u32 scroll;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Demo_Effect/func_80972344.s")
+    scroll = globalCtx->gameplayFrames;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Demo_Effect/func_809723C4.s")
+    OPEN_DISPS(globalCtx->state.gfxCtx, "../z_demo_effect.c", 2737);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Demo_Effect/func_809726AC.s")
+    if (!DemoEffect_CheckCsAction(this, globalCtx, 2)) {
+        if (gSaveContext.entranceIndex == 0x00A0) {
+            if (gSaveContext.sceneSetupIndex == 4) {
+                if (globalCtx->csCtx.frames <= 680) {
+                    func_80078914(&this->actor.projectedPos, NA_SE_EV_GOD_FLYING - SFX_FLAG);
+                }
+            } else {
+                func_80078914(&this->actor.projectedPos, NA_SE_EV_GOD_FLYING - SFX_FLAG);
+            }
+        } else {
+            func_80078914(&this->actor.projectedPos, NA_SE_EV_GOD_FLYING - SFX_FLAG);
+        }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Demo_Effect/func_8097273C.s")
+        gSPSegment(POLY_XLU_DISP++, 8,
+                   Gfx_TwoTexScroll(globalCtx->state.gfxCtx, 0, (scroll * 4) % 512, 0, 128, 64, 1, (scroll * 2) % 256,
+                                    512 - ((scroll * 70) % 512) - 1, 64, 32));
+        gSPSegment(POLY_XLU_DISP++, 9,
+                   Gfx_TwoTexScroll(globalCtx->state.gfxCtx, 0, 0, 0, 16, 96, 1, (scroll * 10) % 256,
+                                    256 - ((scroll * 30) % 512) - 1, 8, 32));
+        gDPSetPrimColor(POLY_XLU_DISP++, 128, 128, this->primXluColor[0], this->primXluColor[1],
+                        this->primXluColor[2], 255);
+        gDPSetEnvColor(POLY_XLU_DISP++, this->envXluColor[0], this->envXluColor[1], this->envXluColor[2], 255);
+        func_80093D84(globalCtx->state.gfxCtx);
+        Matrix_Push();
+        gSPMatrix(POLY_XLU_DISP++, Matrix_NewMtx(globalCtx->state.gfxCtx, "../z_demo_effect.c", 2801),
+                  G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
+        gSPDisplayList(POLY_XLU_DISP++, godLgtTrail);
+        func_80093D18(globalCtx->state.gfxCtx);
+        func_8002EBCC(&this->actor, globalCtx, 0);
+        Matrix_Pull();
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Demo_Effect/func_809727B0.s")
+        this->godLgt.rotation++;
+        if (this->godLgt.rotation > 120) {
+            // if block necessary to match
+            if (1) {
+                this->godLgt.rotation = 0;
```

Comment:
the if 1 cant be on its own line?

Reply:
I tried that before and it did not work. I tried a different permutation by putting the if(1) {} afterward and it still matches.